File: output/CodeLlama/avatar/Python/Java/codeforces_313_B.java
import java.util.Scanner;
public class codeforces_313_B {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String soz = sc.nextLine();
        int[] a = new int[soz.length() + 1];
        for (int i = 1; i < soz.length(); i++) {
            a[i] = a[i - 1];
            if (soz.charAt(i - 1) == soz.charAt(i)) {
                a[i] += 1;
            }
        }
        int n = sc.nextInt();
        int[] arr = new int[n];
        for (int i = 0; i < n; i++) {
            String[] s = sc.nextLine().split(" ");
            arr[i] = a[Integer.parseInt(s[1]) - 1] - a[Integer.parseInt(s[0]) - 1];
        }
        for (int i : arr) {
            System.out.println(i);
        }
    }
}

File: output/StarCoder/avatar/Java/Python/codeforces_313_B.py
import java.io.* ; import java.util.StringTokenizer ; class codeforces_313_B ( ) :
    def __init__ ( self ) :
        self.br = BufferedReader ( )
        self.pw = PrintWriter ( )
    def sol ( self ) :
        s = self.br.readLine ( )
        cum = [ 0 ] * ( len ( s ) + 1 )
        cum [ 0 ] = cum [ len ( s ) ] = 0
        for i in range ( 1, len ( s ) ) :
            if ( s.charAt ( i ) == s.charAt ( i - 1 ) ) :
                cum [ i ] = cum [ i - 1 ] + 1
            else :
                cum [ i ] = cum [ i - 1 ]
        q = Integer.parseInt ( self.br.readLine ( ) )
        while ( q --!= 0 ) :
            l, r = map ( int, self.br.readLine ( ).split ( ) )
            self.pw.println ( cum [ r ] - cum [ l - 1 ] )
    def __del__ ( self ) :
        self.br.close ( )
        self.pw.close ( )
if __name__ == '__main__' :
    c = codeforces_313_B ( )
    c.sol ( )